prboom2/src/r_draw.c

```diff
@@ -107,20 +107,24 @@ static const byte   *tempfuzzmap;
 //#define FUZZOFF (SCREENWIDTH)
 #define FUZZOFF 1
 
-static const int fuzzoffset_org[FUZZTABLE] = {
-  FUZZOFF,-FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,
-  FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,
-  FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,
-  FUZZOFF,-FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,
-  FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,-FUZZOFF,FUZZOFF,
-  FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,
-  FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF
-};
-
-static int fuzzoffset[FUZZTABLE];
-
 static int fuzzpos = 0;
 
+// Lovey01 04/29/2023: Scaled software fuzz
+static int fuzzcellsize;
+
+// NOTE: Formula is (6 + 21 - (9 - i) * (9 - i) * 5 / 19), where i is the fuzz
+// intensity
+// NOTE2: Not const because it's stored closer to fuzzpos this way
```

Comment:
Make it const. As far as I know, zero-initialized data is stored in a separate block from non-zero values, const or otherwise. But, even if that isn't the case, we aren't about to make decisions based on assumptions of where addresses are going to end up. That's a level of fragility that would be a new record even for the doom source.

Rather than violating the meaning of the data, you could instead package things together in a way that satisfies your concern while creating a clean interface that communicates purpose, something like this:
```c
typedef struct {
  int pos;
  int cell_size;
  const byte cmaps[FUZZTABLE];
} fuzz_t;

static fuzz_t fuzz = ...
```

Reply:
Oh yeah you're right, forgot about bss.
Idk what your code preferences for new code is, especially when it concerns efficiency. I can make it const real quick

Reply:
Humans are very bad at these kinds of low-level optimizations and are more likely to work against the compiler. The number one thing is for code to clearly communicate its purpose to the reader / compiler. Concerns to keep in mind are things like algorithmic complexity or high memory consumption. Otherwise something only needs to be optimized by a human if it is a demonstrable bottleneck.

Reply:
I usually try to optimize things too early, it's a bad habit I have. Thanks for the advice